### ISSUE TITLE
Bug: OutOfMemory.off error handling

### ISSUE THREAD

**Issue by mwestphal** [assimp/assimp#5173]
**Describe the bug**
The testing data OutOfMemory.off is not handled as expected in Importer.cpp.

`test/models/invalid/readme.txt` states: 
```
OutOfMemory.off - the number of faces is invalid. There won't be
  enough memory so std::vector::reserve() will most likely fail.
  The exception should be caught in Importer.cpp.
```

**To Reproduce**
1. Open OutOfMemory.off with assimp
2. out of memory error, not expected, should be catched

**Comment by turol** [comment]
This is no longer really true on 64-bit machines. What does actually happen currently? If it throws some error then that's acceptable and IMHO better than potentially hanging the machine by attempting to allocate a ludicrous amount of memory.

**Comment by mwestphal** [comment]
It crashes on my system after saturating my RAM since I have ne swap. ON a standard swap system it would probably hang for a while

**Comment by turol** [comment]
Yeah so need an explicit check somewhere. I added a bunch of these a few years ago (since they caused misbehavior in Valgrind/ASAN) but we clearly need more of them

**Comment by kimkulling** [comment]
Good point, I will have a look.

**Comment by kimkulling** [comment]
I have checked to import the file with the command line tool. It told me that the face declaration is invalid. All exceptions were handled correctly. I have used the following configuration:
- Windows 10
- Vidual Studio 2019
- Asset-Importer-Lib latest master.

Have I missed something? It looked ok for me.

**Comment by mwestphal** [comment]
I'm using linux, no idea what Windows do here.

**Comment by turol** [comment]
Same behavior on Linux.

**Comment by mwestphal** [comment]
Do you mean you did not reproduce on linux @turol ?

**Comment by turol** [comment]
Gives the same error (incorrect number of faces) as on windows.

**Comment by mwestphal** [comment]
Then why does the readme states that the error should be catched ?

**Comment by tellypresence** [comment]
Sounds like a documentation bug?

**Comment by mwestphal** [comment]
That is my understanding, yes. Ideally the error should be caught if possible.
